src/shared/parameter_v2/config/example.yaml

```diff
@@ -0,0 +1,75 @@
+# Examples of all supported Parameter types and their schemas
+--- # Metadata for this config
+metadata:
+  # If shared is true, all parameters in this file will be visible in configs at the same
+  # level in the directoy structure, as well as all children.
+  shared: false
+  # The name of this config. This will be the name used in the generated code
+  name: ExampleConfig
+--- # Parameters
+BoolParam:
+  type: bool
+  value: true
+  description: >
+    Can be true or false
+DoubleParam:
+  type: double # could also be float
```

Comment:
What does 'could be a float'. mean in this case? Like it might as well be a float or it doesn't really matter

Reply:
Hmm, one potential thought is that we can't have `double`'s in firmware here.

Reply:
"Could be a float" was me being lazy and pointing out we could support other floating-point types like `float`. 

Gareth is also right. Easiest thing I can think of is to enforce that anything generated for C must be defined as a float, and fail generation if we're trying to put a double there. This is probably better than auto-converting `doubles` to `floats` if they're being generated for C. Thoughts?

Reply:
Yeah, I think the easiest option is just to have a rule that says that no shared parameters may be `double`'s.

Reply:
As suggested in another comment, I've now explicitly listed all valid parameter types. I included a note that `doubles` will not be generated for C in the description.